dgfit/old/plot_dustmodel.py

```diff
@@ -7,32 +8,22 @@
 from dgfit.obsdata import ObsData
 from dgfit.dustmodel import DustModel
 
-if __name__ == "__main__":
 
+def main():
     # commandline parser
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "--obsdata", help="transform to observed data grids", action="store_true"
+        "obsdata",
+        type=str,
+        default="none",
+        help="transform to observed data grids, with the name of the observed data file as input",
     )
     parser.add_argument("--png", help="save figure as a png file", action="store_true")
     parser.add_argument("--eps", help="save figure as an eps file", action="store_true")
     parser.add_argument("--pdf", help="save figure as a pdf file", action="store_true")
     args = parser.parse_args()
 
-    path = "dgfit/data/mw_rv31"
-    OD = ObsData(
-        [
-            "%s/MW_diffuse_Gordon09_band_ext.dat" % path,
-            "%s/MW_diffuse_Gordon09_iue_ext.dat" % path,
-            "%s/MW_diffuse_Gordon09_fuse_ext.dat" % path,
-        ],
-        "%s/MW_diffuse_Gordon09_avnhi.dat" % path,
-        "%s/MW_diffuse_Jenkins09_abundances.dat" % path,
-        "%s/MW_diffuse_Compiegne11_ir_emission.dat" % path,
-        "%s/dust_scat.dat" % path,
-        ext_tags=["band", "iue", "fuse"],
-        scat_path="%s/Scat_Data/" % path,
-    )
+    OD = ObsData(args.obsdata)
```

Comment:
Should add a check that args.obsdata is not set to "none".  In this case the code will fail.  Better to have an if statement and then raise an error.  https://docs.python.org/3/tutorial/errors.html

Reply:
Given this code moved old, no change needed.